tango_ros_common/tango_ros_native/src/tango_ros_node.cpp

```diff
@@ -821,48 +1039,45 @@ void TangoRosNode::StopPublishing() {
     if (publish_device_pose_thread_.joinable()) {
       if (!tango_data_available_ || !publish_pose_on_tf_
           || !publish_pose_on_topic_) {
-        std::unique_lock<std::mutex> lock(pose_available_mutex_);
         pose_available_.notify_all();
-        pose_available_mutex_.unlock();
       }
       publish_device_pose_thread_.join();
     }
     if (publish_pointcloud_thread_.joinable()) {
       if (!tango_data_available_ ||
           point_cloud_publisher_.getNumSubscribers() <= 0) {
-        std::unique_lock<std::mutex> lock(point_cloud_available_mutex_);
         point_cloud_available_.notify_all();
-        point_cloud_available_mutex_.unlock();
       }
       publish_pointcloud_thread_.join();
     }
     if (publish_laserscan_thread_.joinable()) {
       if (!tango_data_available_ ||
           laser_scan_publisher_.getNumSubscribers() <= 0) {
-        std::unique_lock<std::mutex> lock(laser_scan_available_mutex_);
         laser_scan_available_.notify_all();
-        laser_scan_available_mutex_.unlock();
       }
       publish_laserscan_thread_.join();
     }
     if (publish_fisheye_image_thread_.joinable()) {
       if (!tango_data_available_
           || fisheye_camera_publisher_.getNumSubscribers() <= 0) {
-        std::unique_lock<std::mutex> lock(fisheye_image_available_mutex_);
         fisheye_image_available_.notify_all();
-        fisheye_image_available_mutex_.unlock();
       }
       publish_fisheye_image_thread_.join();
     }
     if (publish_color_image_thread_.joinable()) {
       if (!tango_data_available_
           || color_camera_publisher_.getNumSubscribers() <= 0) {
-        std::unique_lock<std::mutex> lock(color_image_available_mutex_);
         color_image_available_.notify_all();
-        color_image_available_mutex_.unlock();
       }
       publish_color_image_thread_.join();
     }
+    if (publish_mesh_marker_thread_.joinable()) {
+      if (!tango_data_available_
+          || mesh_marker_publisher_.getNumSubscribers() <= 0) {
+        mesh_available_.notify_all();
+      }
+      publish_mesh_marker_thread_.join();
+    }
```

Comment:
Could be useful in the future to combine all thread-related objects in a class: condition_variable, boolean data_available, thread and mutex. This can avoid typos and will probably also reduce number of lines of code.